### DISCUSSION TITLE
Correct way to create random dates with luxon and fakerjs

### DISCUSSION THREAD

**Discussion by fnoquiq** [adonisjs/core#1252] [Help] [answered]
How to create a random date using `luxon` and `fakerjs` in factories?

```ts
import Factory from '@ioc:Adonis/Lucid/Factory'
import { DateTime } from 'luxon'

import Holiday from 'App/Models/Holiday'

export default Factory.define(Holiday, ({ faker }) => {
  return {
    date: DateTime.fromObject({ ordinal: faker.random.number(364) + 1 }),
  }
})
```

This form does not work for leap years. How to make it work for any date?

Note: Adonis v5

**Comment by RomainLanz** [comment, marked as answer]
Have you tried to directly use [ChanceJS `date` function](https://chancejs.com/time/date.html)?

```ts
chance.date()
chance.date({ string: true })
chance.date({ string: true, american: false })
chance.date({ year: 1983 })
```

**Reply by thetutlage** [reply]
@RomainLanz We are using `faker.js` instead of ChanceJS. :). I don't have a great answer to it right now. Will get back to this thread.

**Reply by fnoquiq** [reply]
Can I export many libs from here? If yes, that would be very interesting!

```ts
export default Factory.define(Holiday, ({ fakerjs, chancejs, ...others }) => {
  return {}
})
```

**Reply by RomainLanz** [reply]
You are in a JavaScript file, you can freely install any packages and require them at the top of your file.

**Comment by jjdechavez** [comment]
You can achieve using `fromJSDate()` method from luxon; then pass the faker date on the argument

```ts
import factory from '@adonisjs/lucid/factories'
import { DateTime } from 'luxon'

import Holiday from '#models/holiday'

export const HolidayFactory = factory.define(Holiday, ({ faker }) => {
  return {
    date: DateTime.fromJSDate(faker.date.anytime()),
  }
}).build()
```
